tkg/managementcomponents/management_component_install.go

```diff
@@ -345,21 +351,38 @@ func InstallKappController(clusterClient clusterclient.Client, kappControllerOpt
 	return nil
 }
 
-// InstallManagementPackages installs TKG management packages to the cluster
-func InstallManagementPackages(pkgClient packageclient.PackageClient, mpro ManagementPackageRepositoryOptions) error {
+// InstallManagementPackagesRepo installs TKG management packages repository to the cluster
+func InstallManagementPackagesRepo(pkgClient packageclient.PackageClient, mpro ManagementPackageRepositoryOptions, backoff wait.Backoff) error {
 	// install management package repository
-	err := installManagementPackageRepository(pkgClient, mpro)
-	if err != nil {
-		return errors.Wrap(err, "unable to install management package repository")
-	}
+	return retry.OnError(backoff,
+		func(err error) bool {
+			if err != nil {
+				log.Warning(err.Error() + ", retrying")
+				return true
+			}
+			return false
+		},
+		func() error {
+			return installManagementPackageRepository(pkgClient, mpro)
+		})
+}
 
+// InstallManagementPackages installs TKG management packages to the cluster
+func InstallManagementPackages(pkgClient packageclient.PackageClient, mpro ManagementPackageRepositoryOptions, backoff wait.Backoff) error {
 	// install tkg composite management package
-	err = installTKGManagementPackage(pkgClient, mpro)
-	if err != nil {
-		return errors.Wrap(err, "failure while installing TKG management package")
-	}
-
-	return nil
+	return retry.OnError(backoff,
+		func(err error) bool {
+			// the reconciliation error is tolerable and the process will continue to wait when the
```

Comment:
nit: the comment is a little confusing. 
I think the idea to tolerate certain errors by retrying _until_ a reconciliation error is seen, right?

suggestion:
// the error is tolerable, so continue with retry until a reconciliation error is returned

Reply:
Thanks @vuil ! That is much clearer. I updated this part.